src/target/sam3x.c

```diff
@@ -579,9 +582,9 @@ static int sam_flash_write(target_flash_s *f, target_addr_t dest, const void *sr
 
 	target_mem_write(t, dest, src, len);
 	if(sam_flash_cmd(t, base, sf->write_cmd, chunk))
```

Comment:
As this if is immediately followed by returns on both branches, we'd suggest it should be replaced by a direct `return !sam_flash_cmd();` (clang-tidy should say something to this effect too)